jax/experimental/optimizers.py

```diff
@@ -94,7 +107,28 @@
     lambda xs: ((xs.packed_state,), (xs.tree_def, xs.subtree_defs)),
     lambda data, xs: OptimizerState(xs[0], data[0], data[1]))
 
-def optimizer(opt_maker):
+
+Array = jnp.ndarray
```

Comment:
We use `Array = Any` as a placeholder, because in JAX array arguments are also called by tracer objects during JIT compilation; see https://github.com/google/jax/blob/4236eb2b5929b9643977553f7f988ca518b7df4e/jax/ad_util.py#L23

Reply:
Thanks.